index.bs

```diff
@@ -253,18 +253,251 @@ Control Protocols {#control}
+individually to controllers that have requested availability for the URLs that
+have changed in availability state within the watch duration of the original
+availability request.
+
+
+To save power, the controller may disconnect the QUIC connection and
+later reconnect to send availablity requests and receive availability
+responses and updates.
+
+
+To start a presentation, the controller may send a
+presentation-start-request message to the receiver with the following
+values:
+
+- presentation-id: the presentation identifier
+
+- url: the selected presentation URL
+
+- headers: headers that the receiver should use to fetch the
+presentationUrl.  For example, section 6.6.1 of
+[[PRESENTATION-API|Presentation API]] says that the Accept-Language
+header should be provided.
+
+The presentation ID must follow the restrictions defined by
+[[PRESENTATION-API|Presentation API]] section 6.1, in that it must
+consist of at least 16 ASCII characters.
+
+
+When the receiver receives the presentation-start-request, it should send back a
+presentation-start-response message after either the presentation URL has been
+feteched and loaded or the receiver has failed to do so. If it has failed, it
+must respond with the appropriate result (such as invalid-url or timeout).  If
+it has succeeded, it must reply with a success result.  Additionally, the
+response must include the following:
+
+- connection-id: An ID that both endpoints can use to send connection messages
+to each other.  It is chosen by the receiver for ease of implementation (if the
+message receiver chooses the connection-id, it may keep the ID unique across
+connections, thus making message demuxing/routing easier).
+
+<!-- TODO: Add optional HTTP response code to the response? -->
+
+To send a presentation message, the controller or receiver may send a
+presentation-connection-message with the following values:
+
+- connection-id: The ID from the presentation-start-response or
+presentation-connection-open-response messages.
+
+- message: the presentation message data.  
+
+
+To terminate a presentation, the controller may send a
+presentation-termination-request message with the following values:
+
+- presentation-id: The ID of the presentation to terminate.
+
+- reason: The reason the presentation is being terminated.
+
+
+When a receiving endpoint receives a presentation-termination-request, it should
+send back a presentation-termination-response message to the requesting
+endpoint.  It may also notify other controllers about the termination by sending
+a presentation-termination-event message.  And it can send the same message if
+it terminates a presentation without a request from a controller to do so. This
+message contains the following values:
+
+- presentation-id: The ID of the presentation that was terminated.
+
+- reason: The reason the presentation was terminated.
+
+<!-- TODO: Split up reason into reason and whether it was triggered by the user
```

Comment:
Not required by the API spec, if I remember, but probably fine to include that level of detail in the protocol.

Reply:
It was something you asked for :).